pallets/foreign-investments/src/lib.rs

```diff
@@ -242,168 +313,56 @@ pub mod pallet {
 	#[pallet::storage]
-	#[pallet::getter(fn token_swap_order_ids)]
-	pub(super) type TokenSwapOrderIds<T: Config> = StorageDoubleMap<
-		_,
-		Blake2_128Concat,
-		T::AccountId,
-		Blake2_128Concat,
-		T::InvestmentId,
-		T::TokenSwapOrderId,
-	>;
-
-	/// Maps an investor and their `InvestmentId` to the collected investment
-	/// amount, i.e., the payment amount of pool currency burned for the
-	/// conversion into collected amount of tranche tokens based on the
-	/// fulfillment price(s).
-	///
-	/// NOTE: The lifetime of this storage starts with receiving a notification
-	/// of an executed investment via the `CollectedInvestmentHook`. It ends
-	/// with transferring the collected tranche tokens by executing
-	/// `notify_executed_collect_invest` which is part of
-	/// `collect_foreign_investment`.
-	#[pallet::storage]
-	pub type CollectedInvestment<T: Config> = StorageDoubleMap<
-		_,
-		Blake2_128Concat,
-		T::AccountId,
-		Blake2_128Concat,
-		T::InvestmentId,
-		CollectedAmount<T::Balance>,
-		ValueQuery,
-	>;
-
-	/// Maps an investor and their `InvestmentId` to the collected redemption
-	/// amount, i.e., the payment amount of tranche tokens burned for the
-	/// conversion into collected pool currency based on the
-	/// fulfillment price(s).
-	///
-	/// NOTE: The lifetime of this storage starts with receiving a notification
-	/// of an executed redemption collection into pool currency via the
-	/// `CollectedRedemptionHook`. It ends with having swapped the entire amount
-	/// to foreign currency which is assumed to be asynchronous.
-	#[pallet::storage]
-	pub type CollectedRedemption<T: Config> = StorageDoubleMap<
-		_,
-		Blake2_128Concat,
-		T::AccountId,
-		Blake2_128Concat,
-		T::InvestmentId,
-		CollectedAmount<T::Balance>,
-		ValueQuery,
-	>;
-
-	/// Maps an investor and their investment id to the foreign payment currency
-	/// provided on the initial investment increment.
-	///
-	/// The lifetime is synchronized with the one of
-	/// `InvestmentState`.
-	#[pallet::storage]
-	pub type InvestmentPaymentCurrency<T: Config> = StorageDoubleMap<
-		_,
-		Blake2_128Concat,
-		T::AccountId,
-		Blake2_128Concat,
-		T::InvestmentId,
-		T::CurrencyId,
-		ResultQuery<Error<T>::InvestmentPaymentCurrencyNotFound>,
-	>;
-
-	/// Maps an investor and their investment id to the foreign payout currency
-	/// requested on the initial redemption increment.
-	///
-	/// The lifetime is synchronized with the one of
-	/// `RedemptionState`.
-	#[pallet::storage]
-	pub type RedemptionPayoutCurrency<T: Config> = StorageDoubleMap<
-		_,
-		Blake2_128Concat,
-		T::AccountId,
-		Blake2_128Concat,
-		T::InvestmentId,
-		T::CurrencyId,
-		ResultQuery<Error<T>::RedemptionPayoutCurrencyNotFound>,
-	>;
+	pub(super) type ForeignIdToSwapId<T: Config> =
+		StorageMap<_, Blake2_128Concat, ForeignId<T>, T::SwapId>;
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		ForeignInvestmentUpdated {
```

Comment:
@wischli I'm not sure if these events apply now. I think not because `pallet-order-book` and `pallet-investments` will already emit their corresponding events when the "state" changes.

Reply:
Yes, you can certainly can drop all events which are already handled by the Orderbook and Investments! AFAIK, none of the events are used right now. I would add events for mutations of the `ForeignInvestmentInfo` and `ForeignRedemptionInfo`.